SW4STM32/BellaLui/Application/HostBoard/Src/misc/state_machine.c

```diff
@@ -222,31 +175,24 @@ void TK_state_machine(void const *argument) {
 		case STATE_SECONDARY: {
 			uint8_t counterTdTrig = 0;
 
-			// if a given time has passed since the last time the check was done, do the check
-			if ((HAL_GetTick() - time_tmp) > TOUCHDOWN_DELAY_TIME) {
-				if (baroIsReady) {
-					// save time of check
-					time_tmp = HAL_GetTick();
-
-					// check if the altitude hasn't varied of more than a given amount since last time
-					if (abs_fl32(baro_data->altitude - td_last_alt) > TOUCHDOWN_ALT_DIFF) {
-						// if the altitude difference is still large, this means the rocket is descending and so the touch down counter is kept to zero
-						td_counter = 0;
-					} else {
-						// if the altitude difference is in bounds, the counter is incremented
-						td_counter++;
-						if (td_counter > TOUCHDOWN_BUFFER_SIZE) {
-							// if the counter is larger than a given value, toggle the state trigger
-							counterTdTrig = 1;
-						}
-					}
-					td_last_alt = baro_data->altitude;
+			const float baro_data_altitude = baroIsReady ? baro_data->altitude : -1;
 
-					if (counterTdTrig) {
+			uint8_t state_secondary_status = state_machine_helpers::handleSecondaryState(HAL_GetTick(), time_tmp, baroIsReady, baro_data_altitude, td_last_alt, td_counter);
+
+			if(state_primary_status != 0){
```

Comment:
you probably meant "state_secondary_status"

Reply:
nice catch